src/gui/newwizard/SetupWizardWindow.cpp

```diff
@@ -0,0 +1,152 @@
+#include "SetupWizardWindow.h"
+
+#include <QLabel>
+
+#include "Pagination.h"
+#include "ui_SetupWizardWindow.h"
+
+namespace OCC::Wizard {
+
+SetupWizardWindow::SetupWizardWindow(QWidget *parent)
+    : QDialog(parent)
+    , _ui(new ::Ui::SetupWizardWindow)
+    //    , _transitionLayout(new QHBoxLayout)
+    , _transitionProgressIndicator(new QProgressIndicator)
+{
+    _ui->setupUi(this);
+
+    // make sure to initialize these after UI has been set up
+    _pagination.reset(new Pagination(_ui->paginationLayout));
+
+    slotHideErrorMessageWidget();
+
+    _contentLayout.reset(new QStackedLayout(_ui->contentWidget));
+    _contentLayout->setAlignment(Qt::AlignCenter);
+
+    connect(_ui->cancelButton, &QPushButton::clicked, this, &SetupWizardWindow::reject);
+
+    // set up transition
+    //    _transitionLayout->addWidget(_transitionProgressIndicator.get());
+
+    connect(_ui->nextButton, &QPushButton::clicked, this, [this]() {
+        slotStartTransition();
+        emit nextButtonClicked(_pagination->activePageIndex());
+    });
+
+    connect(_ui->backButton, &QPushButton::clicked, this, [this]() {
+        slotStartTransition();
+        emit backButtonClicked(_pagination->activePageIndex());
+    });
+
+    connect(_pagination.get(), &Pagination::paginationEntryClicked, this, [this](PageIndex clickedPageIndex) {
+        slotStartTransition();
+        emit paginationEntryClicked(_pagination->activePageIndex(), clickedPageIndex);
+    });
+
+    _transitionProgressIndicator->setFixedSize(32, 32);
+}
+
+void SetupWizardWindow::displayPage(AbstractSetupWizardPage *page, PageIndex index)
+{
+    // save some CPU cycles
+    _transitionProgressIndicator->stopAnimation();
```

Comment:
Is this needed, the default state should be off, also as far as I know all logic here is in the painter so once its hidden there are no cycles to save.

Reply:
I wasn't sure whether to use it or not. I thought, since it's offered, why not use it, it definitely can't hurt... I'll check that again.